src/psyclone/psyad/transformations/preprocess.py

```diff
@@ -95,6 +95,10 @@ def preprocess_trans(kernel_psyir, active_variable_names):
         else:
             sym_maths = SymbolicMaths.get()
             sym_maths.expand(assignment.rhs)
+            # The expand function does not expand if one of the terms
+            # is a literal e.g. 0.5*(a+b) so run the associativity
+            # function to catch any such cases.
```

Comment:
Is this really true? I tried:
![image](https://user-images.githubusercontent.com/17409212/176722292-bb3b5494-faec-47bf-903e-45a6f9f03f8f.png)

Reply:
You're quite right. sympy in Python does what is expected. I need to find out why it is not working via PSyclone.

Reply:
The problem was to do with a bug in the PSyclone expand function. I've fixed this now.